### PR TITLE
37 implement sphinx for docs

### PR DESCRIPTION
**Pull request by HTJense**
This should:

- Add a docs target in tox.
- Add a workflow such that docs are built whenever the `master` branch gets pushed or a PR gets merged.

It should also be setup for readthedocs, but I am not 100% sure if it works correctly.

### CODE REVIEW

**Comment by HTJense** [comment]
Current state passes Documentation checks (see [here](https://github.com/simonsobs/SOLikeT/actions/runs/4282445705/jobs/7456730996)).

**Comment by HTJense** [comment]
This seems to be setup fine now (see [the current docs](https://soliket.readthedocs.io/en/latest/index.html)).
One (optional) improvement we can consider is to use autodoc's feature to [fake imports](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_mock_imports) so that pyCCL (and others) installation can be skipped for speed and stability issues.

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+**SO Likelihoods and Theories**
+
+A centralized package for likelihood and theory implementations for SO.
+
+
+Installation
+============
+
+For a set of detailed instructions for different machines (e.g. NERSC), please see `here<INSTALL.rst>`_.
```
Comment (itrharrison) [inline README.rst:16]: Link formatting problem.

**Review by itrharrison — CHANGES_REQUESTED** [review]

File: README.rst (hunk 1 above)
Comment (HTJense) [reply README.rst]: Fixed.

**Review by itrharrison — APPROVED** [review]
LGTM!

**Pull request merged**